tests/test_lit_server.py

```diff
@@ -119,3 +121,14 @@ def test_run():
     assert '{"output":16.0}' in output
     os.remove("client.py")
     process.kill()
+
+
+@pytest.mark.asyncio()
+async def test_stream(simple_stream_api):
+    server = LitServer(simple_stream_api, accelerator="cpu", stream=True)
+    expected_output = "prompt=Hello World generated_output=LitServe is streaming output".lower().replace(" ", "")
+
+    async with LifespanManager(server.app) as manager, AsyncClient(app=manager.app, base_url="http://test") as ac:
+        response = await ac.post("/stream-predict", json={"prompt": "Hello World"}, timeout=10)
```

Comment:
Can we test individual streaming steps, like wouldn't the test pass even if streaming was not happening?

Reply:
added a test to assert all functions are called correctly by unit testing the [streaming_loop](https://github.com/Lightning-AI/litserve/pull/37/files#diff-274978086d8b4f049cffd0a53c0a2b1634c323f7ae581e87427c3b0c9046716aR153)